src/Sentry/Internal/FileSystem.cs

```diff
@@ -16,38 +19,112 @@ public IEnumerable<string> EnumerateFiles(string path, string searchPattern) =>
     public IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption) =>
         Directory.EnumerateFiles(path, searchPattern, searchOption);
 
-    public void CreateDirectory(string path) => Directory.CreateDirectory(path);
+    public DirectoryInfo? CreateDirectory(string path)
+    {
+        if (!_options?.DisableFileWrite is false)
+        {
+            _options?.LogDebug("Skipping creating directory. Writing to file system has been explicitly disabled.");
+            return null;
+        }
+
+        return Directory.CreateDirectory(path);
+    }
+
+    public bool? DeleteDirectory(string path, bool recursive = false)
```

Comment:
I don't think this method can ever return null... so could be `bool` return type.